### ISSUE TITLE
Spectrograph number missing from cframe filepath returned by desispec.io.meta.findfile?

### ISSUE THREAD

**Issue by michaelJwilson** [desihub/desispec#779]
```findfile('cframe', night=20200317, expid=18, nside=64, camera='r', spectrograph=9, rawdata_dir=None, specprod_dir=None, download=False, outdir=None, qaprod_dir=None)```

returns:

```/global/cscratch1/sd/mjwilson/minitest-19.2/spectro/redux/mini/exposures/20200317/00000018/cframe-r-00000018.fits```

even though spectrograph is provided.  Each camera band, in this case r, should be accompanied by spectrograph number.  In this case, real path ends with ```cframe-r1-00000018.fits``` to ```cframe-r9-00000018.fits```.

It might make sense to add a file exists check before returning the filepath?

**Comment by weaverba137** [comment]
> It might make sense to add a file exists check before returning the filepath?

Not necessarily.  There may be cases where this function is used to generate a filename that will be written to in the future.

**Comment by sbailey** [comment]
@michaelJwilson I opened PR #780 to address this.  The previous intended usage is `findfile(..., camera='r9')`, i.e. the "camera" includes both the spectrograph channel (b,r,z) and the spectrograph number (0-9).  The way you were trying to use this is reasonable though, so I added support for that.  In general most places in the code that refer to "camera" mean b0,r1..z9, and we use "arm" or "channel" when referring to just b,r,z without the spectrograph number, so for consistency it would be better for your code to also use camera meaning b0, r1, ... z9.

And as @weaverba137 noted, we also use `findfile` to determine where new files should be written, so we don't do an existence check as part of `findfile` itself.
